--- dotlock.c.orig	2015-12-01 01:07:02.212643687 +0100
+++ dotlock.c	2015-12-01 01:07:46.746899721 +0100
@@ -412,7 +412,7 @@
   for (;;)
     {
 #ifndef HAVE_DOSISH_SYSTEM
-# ifndef __riscos__
+# if 0 /*ndef __riscos__
       if ( !link(h->tname, h->lockname) )
         {
           /* fixme: better use stat to check the link count */
@@ -425,7 +425,7 @@
           return -1;
 	}
 # else /* __riscos__ */
-      if ( !renamefile(h->tname, h->lockname) )
+      if ( !rename(h->tname, h->lockname) )
         {
           h->locked = 1;
           return 0; /* okay */
@@ -562,7 +562,7 @@
       return -1;
     }
 
-#ifndef __riscos__
+# if 0 /*ndef __riscos__
   if ( unlink( h->lockname ) )
     {
       log_error ("release_dotlock: error removing lockfile `%s'\n",
@@ -572,7 +572,7 @@
   /* Fixme: As an extra check we could check whether the link count is
      now really at 1. */
 #else /* __riscos__ */
-  if ( renamefile (h->lockname, h->tname) )
+  if ( rename (h->lockname, h->tname) )
     {
       log_error ("release_dotlock: error renaming lockfile `%s' to `%s'\n",
                  h->lockname, h->tname);
